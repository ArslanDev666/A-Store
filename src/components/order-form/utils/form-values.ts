import { OrderDeliveryType } from 'types/order';

export enum DeliveryValuesEnum {
  /**
   * Доставка по России
   */
  DeliveryInRussia = 'deliveryInRussia',
  /**
   * Курьером по Москве
   */
  CourierInMoscow = 'courierInMoscow',
  /**
   * Самовывоз
   */
  Pickup = 'pickup',
}

export type DeliveryValueType = {
  /**
   * Текст для блока с итоговой ценой
   */
  text: string;
  /**
   * Цена доставки
   */
  price: number;
  /**
   * Текст для блока с формой
   */
  label: OrderDeliveryType;
};

export type DeliveryValuesType = Record<DeliveryValuesEnum, DeliveryValueType>;

export const deliveryValues: DeliveryValuesType = {
  deliveryInRussia: {
    text: 'Доставка по России — ',
    price: 350,
    label: 'Доставка по России — 350₽',
  },
  courierInMoscow: {
    text: 'Курьером по Москве —',
    price: 300,
    label: 'Курьером по Москве — 300₽',
  },
  pickup: {
    text: 'Самовывоз (пр-т Андропова, 18 корп. 3)',
    price: 0,
    label: 'Самовывоз (пр-т Андропова, 18 корп. 3)',
  },
};
