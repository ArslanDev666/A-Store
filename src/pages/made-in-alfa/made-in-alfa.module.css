.root {
  padding: var(--gap-3xl) 0 var(--gap-6xl);
}

.products {
  display: grid;
  grid-template-columns: repeat(auto-fit, 370px);
  gap: var(--gap-4xl) var(--gap-xl);
}

@media screen and (max-width: 1200px) {
  .products {
    grid-template-columns: repeat(auto-fit, 300px);

    justify-content: center;
  }
}
@media screen and (max-width: 375px) {
  .products {
    grid-template-columns: repeat(auto-fit, 1fr);
  }
}