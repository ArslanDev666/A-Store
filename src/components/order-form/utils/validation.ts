import * as yup from 'yup';

import { DeliveryValuesEnum } from './form-values';

const PHONE_NUMBER_LENGTH = 18;
const MIN_FULLNAME_LENGTH = 3;

const FIND_LETTERS_REGEX = /^[A-Яа-я\s]+$/g;

const ERRORS = {
  required: 'Пожалуйста, заполните все обязательные поля',
  trim: 'В поле не должно быть пустых символов в начале и в конце строки',
  fullname: 'Введите имя',
  fullnameMin: 'Слишком короткое имя',
  email: 'Введите email',
  phone: 'Введите корректный номер',
  policyAgree: 'Подтвердите согласие на обработку данных',
};

export const validationFormSchema = yup
  .object({
    fullname: yup
      .string()
      .matches(FIND_LETTERS_REGEX, ERRORS.fullname)
      .trim(ERRORS.trim)
      .min(MIN_FULLNAME_LENGTH, ERRORS.fullnameMin)
      .required(ERRORS.required)
      .strict(true),
    email: yup.string().email(ERRORS.email).required(ERRORS.required),
    phone: yup.string().min(PHONE_NUMBER_LENGTH, ERRORS.phone).required(ERRORS.required),
    address: yup.string().when('delivery', {
      is: DeliveryValuesEnum.Pickup,
      then: (schema) => schema.required(ERRORS.required),
    }),
    delivery: yup.string(),
    policyAgree: yup.bool().oneOf([true], ERRORS.policyAgree),
    comment: yup.string(),
    payment: yup.string(),
  })
  .required();

export type FormDataType = yup.InferType<typeof validationFormSchema>;
