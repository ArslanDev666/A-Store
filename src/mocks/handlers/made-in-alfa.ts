import { rest } from 'msw';

import { PRODUCTS_API } from 'utils/const';

import { madeInAlfaProductsMock } from '../data/products';

export const getMadeInAlfaProductsMockHandler = rest.get(
  `${PRODUCTS_API}/made-in-alfa`,
  (_reg, res, ctx) => {
    const data = madeInAlfaProductsMock;

    return res(ctx.status(200), ctx.json(data));
  }
);
