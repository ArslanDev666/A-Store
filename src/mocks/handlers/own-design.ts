import { rest } from 'msw';

import { PRODUCTS_API } from 'utils/products-api';

import { ownDesignProductsMock } from '../data/products';

export const getOwnDesignProductsMockHandler = rest.get(
  `${PRODUCTS_API}/your-design`,
  (_reg, res, ctx) => {
    const data = ownDesignProductsMock;

    return res(ctx.status(200), ctx.json(data));
  }
);
