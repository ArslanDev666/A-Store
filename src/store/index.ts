import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
import { combineReducers, configureStore, PreloadedState } from '@reduxjs/toolkit';
import createSagaMiddleware from 'redux-saga';

import { cartReducer } from './cart';
import { madeInAlfaReducer } from './made-in-alfa';
import { notificationsReducer } from './notifications';
import { ownDesignReducer } from './own-design';
import { productReducer } from './product';
import { rootSaga } from './root-saga';

const sagaMiddleware = createSagaMiddleware();
const middlewares = [sagaMiddleware];

const rootReducer = combineReducers({
  notifications: notificationsReducer,
  ownDesign: ownDesignReducer,
  madeInAlfa: madeInAlfaReducer,
  product: productReducer,
  cart: cartReducer,
});

export const setupStore = (preloadedState?: PreloadedState<RootState>) => {
  return configureStore({
    reducer: rootReducer,
    preloadedState,
    devTools: true,
    middleware: middlewares,
  });
};

export const store = setupStore();

sagaMiddleware.run(rootSaga);

export type RootState = ReturnType<typeof rootReducer>;
export type AppStore = ReturnType<typeof setupStore>;

export const useAppDispatch = () => useDispatch<AppStore['dispatch']>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
