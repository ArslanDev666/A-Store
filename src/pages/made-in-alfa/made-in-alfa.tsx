import React from 'react';

import { Gap } from '@alfalab/core-components/gap';
import { Spinner } from '@alfalab/core-components/spinner';
import { Typography } from '@alfalab/core-components/typography';

import { Product } from 'components/product';
import { Container } from 'components/ui/container';
import { SectionTitle } from 'components/ui/section-title';

import { useAppSelector } from 'store';
import { madeInAlfaIsLoadingSelector, madeInAlfaSelector } from 'store/made-in-alfa';

import { ProductType } from 'types/product';

import styles from './made-in-alfa.module.css';

const TITLE_PAGE = 'Сделано в Альфе';
const DESCRIPTION_PAGE = 'Хотим каждую из этих вещей! Себе, родным и друзьям';

const MadeInAlfaPage = () => {
  const products = useAppSelector(madeInAlfaSelector);
  const isLoading = useAppSelector(madeInAlfaIsLoadingSelector);

  return (
    <div className={styles.root} data-test-id='made-in-alfa-page'>
      <Typography.Title tag='h1' hidden>
        {TITLE_PAGE}
      </Typography.Title>

      <Container>
        <SectionTitle title={TITLE_PAGE} description={DESCRIPTION_PAGE} />
        <Gap size='6xl' />

        <Spinner visible={isLoading} size='m' />

        {!!products.length && (
          <ul className={styles.products}>
            {products.map((product: ProductType) => (
              <li key={product.id}>
                <Product
                  image={product.preview}
                  title={product.title}
                  price={product.price}
                  id={product.id}
                />
              </li>
            ))}
          </ul>
        )}
      </Container>
    </div>
  );
};

export { MadeInAlfaPage };
