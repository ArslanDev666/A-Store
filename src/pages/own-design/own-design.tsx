import React from 'react';

import { Gap } from '@alfalab/core-components/gap';
import { Spinner } from '@alfalab/core-components/spinner';
import { Typography } from '@alfalab/core-components/typography';

import { ProductCategory } from 'components/product-category';
import { Container } from 'components/ui/container';
import { SectionTitle } from 'components/ui/section-title';

import { useAppSelector } from 'store';
import {
  ownDesignIsLoadingSelector,
  ownDesignSelector,
} from 'store/own-design';

import { CategoryType } from 'types/product-category';

import styles from './own-design.module.css';

const TITLE_PAGE = 'Свой дизайн';
const DESCRIPTION_PAGE =
  'Выберите вещь, а затем — цвет, размер и стикер. Перенесём стикер на вещь как на фото';

const OwnDesignPage = () => {
  const products = useAppSelector(ownDesignSelector);
  const isLoading = useAppSelector(ownDesignIsLoadingSelector);

  return (
    <div className={styles.root} data-test-id='own-design-page'>
      <Typography.Title tag='h1' hidden>
        {TITLE_PAGE}
      </Typography.Title>

      <Container>
        <SectionTitle title={TITLE_PAGE} description={DESCRIPTION_PAGE} />
        <Gap size='6xl' />

        <Spinner visible={isLoading} size='m' />

        {!!products.length && (
          <>
            <ul className={styles.categories}>
              {products.map((category: CategoryType) => (
                <ProductCategory
                  title={category.title}
                  description={category.description}
                  products={category.products}
                  id={category.id}
                  key={category.id}
                />
              ))}
            </ul>
            <Gap size='8xl' />
            <div className={styles.footer}>
              <Typography.TitleResponsive
                weight='medium'
                tag='h4'
                color='tertiary'
                view='xsmall'
              >
                Посмотреть и потрогать все стикеры можно в A-Store на
                Технопарке. А ещё там можно добавить сразу несколько стикеров на
                одну вещь.
              </Typography.TitleResponsive>
            </div>
          </>
        )}
      </Container>
    </div>
  );
};

export { OwnDesignPage };
