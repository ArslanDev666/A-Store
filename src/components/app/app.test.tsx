import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import { setupStore, store } from 'store';
import { madeInAlfaActions } from 'store/made-in-alfa';
import { ownDesignActions } from 'store/own-design';

import { renderWithProviders } from 'utils/tests-utils';

import { App } from '.';

describe('snapshot testing', () => {
  it('should render App', () => {
    const component = renderer.create(
      <Provider store={setupStore()}>
        <App />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});

describe('functions testing', () => {
  const mockDispatch = jest.fn();

  beforeEach(() => {
    jest.mock('store/index.ts', () => ({
      useAppDispatch: mockDispatch,
    }));
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render App', () => {
    renderWithProviders(<App />, {
      store: { ...store, dispatch: mockDispatch },
    });

    expect(mockDispatch).toHaveBeenCalledWith(ownDesignActions.request());
    expect(mockDispatch).toHaveBeenCalledWith(madeInAlfaActions.request());
  });
});
