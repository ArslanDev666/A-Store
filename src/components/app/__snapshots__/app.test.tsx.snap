// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing should render App 1`] = `
<div
  className="layout"
  data-test-id="app-layout"
>
  <header
    className="header"
    data-test-id="header"
  >
    <a
      className="logo"
      href="/"
      onClick={[Function]}
    >
      A-Store
    </a>
    <button
      className="sidebarToggle"
      onClick={[Function]}
      type="button"
    >
      <span>
        меню
      </span>
      <svg
        fill="currentColor"
        focusable="false"
        height="24"
        role="img"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M3 13.25h18v-2.5H3v2.5zm0-7.5v2.5h18v-2.5H3zm0 12.5h18v-2.5H3v2.5z"
        />
      </svg>
    </button>
  </header>
  <div
    className="content"
  >
    <div
      className="root"
      data-test-id="home-page"
    >
      <h1
        className="typography__component_a3ydv typography__styrene-medium_1ixkt typography__medium_a3ydv"
        hidden={true}
      >
        A-Store
      </h1>
      <a
        className="card"
        data-test-id="card"
        href="/sdelano-v-alfe"
        onClick={[Function]}
      >
        <img
          alt="Сделано в Альфе"
          className="cardImage"
          src="banner-made-in-alfa.jpeg"
        />
        <div
          className="cardContent"
        >
          <h2
            className="typography__component_a3ydv typography__styrene-medium_1x9yf typography__bold_a3ydv"
          >
            Сделано в Альфе
          </h2>
        </div>
      </a>
      <a
        className="card"
        data-test-id="card"
        href="/svoy-dizain"
        onClick={[Function]}
      >
        <img
          alt="Свой дизайн"
          className="cardImage"
          src="banner-own-design.jpeg"
        />
        <div
          className="cardContent"
        >
          <h2
            className="typography__component_a3ydv typography__styrene-medium_1x9yf typography__bold_a3ydv"
          >
            Свой дизайн
          </h2>
        </div>
      </a>
    </div>
  </div>
  <div
    className="cart"
  />
</div>
`;
